### ISSUE TITLE
[upgrade] 3.0.0 to 3.1.0 upgrade fails for NVIDIA

### ISSUE THREAD

**Issue by LukeShortCloud** [LukeShortCloud/winesapOS#409]
AMD and Intel upgrades work great. The problem with NVIDIA (and other DKMS kernel modules), is that, on the performance image, the Linux kernel is not upgraded. The old kernel was built with GCC 11. After we switch to the latest Arch Linux repositories, GCC 12 is currently installed. It is not backwards compatible so all DKMS modules fail to rebuild. If we update to the latest Arch Linux, the Linux kernel is built with GCC 12 so it is not a problem.

tl;dr? To fix this issue, we need to upgrade the Linux kernels as part of our upgrade process. If we detect that the `nvidia-dkms` package is installed, then we need to re-install it to force the rebuilt of the DKMS for all kernels and the rebuild of the initramfs.

Related: https://github.com/LukeShortCloud/winesapOS/issues/379

**Comment by LukeShortCloud** [comment]
We should actually upgrade all NVIDIA packages (the same ones installed as part of the first-time setup).
